### ISSUE TITLE
When_mapping_a_data_reader_to_a_dto_with_nullable_enum

### ISSUE THREAD

**Issue by AgentFire** [AutoMapper/AutoMapper.Data#47]
https://github.com/AutoMapper/AutoMapper.Data/blob/master/AutoMapper.Data.Tests/DataReaderMapping.cs#L433

The test is essentially rigged (`Map` call is commented out, and the mapping is performed manually) to present successful results.

However, when an actual mapping is attempted, - from `IDataReader` to a DTO with nullable enum member - the following exception is thrown:

```
AutoMapper.AutoMapperMappingException: 'Error mapping types.'
Inner exception:
InvalidCastException: Specified cast is not valid.
```

**Comment by lbargaoanu** [comment]
A PR is welcome.

**Comment by jbogard** [comment]
Fixed by #52
